### PR TITLE
feat(build): add configuration files for VS Code devcontainer

### PR DESCRIPTION
**Pull request by domire8**
## Description

With the suggested configuration, you can build an run tests for the core libraries. If you wanted to build and test clproto at the same time, I'm sure this can be done with one or two changes but this is not the main use case of the devcontainer.

## Review guidelines

Estimated Time of Review: 5 minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "control libraries",
+    "remoteUser": "ros2",
+    "build": {
+        "dockerfile": "../Dockerfile.ci",
+        "context": "..",
+        "target": "development",
+        "args": {
+            "ROS2_VERSION": "humble",
```
Comment (eeberhard) [inline .devcontainer/devcontainer.json:9]: This arg is not used by the CI dockerfile, the base image is hardcoded. On a side-note, it would probably be more correct to use an `ubuntu` base image rather than `ghcr.io/aica-technology/ros2-ws:humble` anyway.

**Review by eeberhard — APPROVED** [review]

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (domire8) [reply .devcontainer/devcontainer.json:9]: Yeah we could use ubuntu but then we end up installing a bunch of apt packages (cmake, make, complier) that we already have in the workspace?

Comment (eeberhard) [reply .devcontainer/devcontainer.json:9]: can't we use the existing development-dependencies image then? I just think it's weird to see ROS stuff here

Comment (domire8) [reply .devcontainer/devcontainer.json:9]: I can do that yeah, I just think that the development dependencies image could be removed at some point because its very heavy and apart from potentially here, it's not used anywhere. `ubuntu` is slightly annoying because we'd have to configure a user again.

**Review by eeberhard — APPROVED** [review]

**Comment by domire8** [comment]
@eeberhard the change of base image just broke everything :smile: The python packages are copied to `/tmp/python-home/${USER}/.local/lib/python3.10/site-packages/` so if the user doesn't correspond between ros images and this image, the copying will fail....

**Comment by eeberhard** [comment]
Ah, I see.. ok, that's more of a change than I initially expected. If it were just some permission / ownership problems it could be solved with a `COPY --chown`, but the python stuff is on a named path...

My main issue with the ros2 base image is that it's not really the responsibility of this repo to know what the downstream users will be. This image could also be used to provide pre-built binaries to other ubuntu users, for example, and then it's the same question about how to copy user-specific paths.

Probably the "right" solution is to put all the build outputs in specific folders. For example, maybe keep just the `.local` folder of python, not the whole python home path. So instead of a straight `COPY from=cl / /` in the downstream dockerfile, you would have to do something like
```
COPY from=cl /apt-deps /
COPY from=cl --chown=${USER}:${USER} /usr /usr
COPY from=cl --chown=${USER}:${USER} /python/.local /home/${USER}/.local
```

> ubuntu is slightly annoying because we'd have to configure a user again

A lot of the user setup in development dependencies was also about permissions and SSH, so in the end it might not be so bad to have a simple user.

I think all that is nice to try at some point to make this image a bit more generic, but it's also not the most important thing.

Anyway TL;DR, it's a bit of pain to do all this and I don't want to enforce a cumbersome process / delay other work; if we are the main users of this image for now, then it's fine to revert back to ros2 base image and keep this idea in the backlog.
